native/src/components/Header.tsx

```diff
@@ -178,14 +186,19 @@ const Header = (props: PropsType): ReactElement => {
           )}
         </HorizontalLeft>
         <MaterialHeaderButtons cancelLabel={t('cancel')} theme={theme}>
-          {!peeking && categoriesAvailable && renderItem(t('search'), 'always', goToSearch, t('search'), 'search')}
-          {!peeking &&
-            goToLanguageChange &&
-            renderItem(t('changeLanguage'), 'always', goToLanguageChange, t('changeLanguage'), 'language')}
-          {showShare && renderItem(t('share'), 'never', onShare, t('share'), undefined)}
-          {showChangeLocation && renderItem(t('changeLocation'), 'never', goToLanding, t('changeLocation'), undefined)}
-          {renderItem(t('settings'), 'never', goToSettings, t('settings'), undefined)}
-          {routeCityModel && renderItem(t('disclaimer'), 'never', goToDisclaimer, t('disclaimer'), undefined)}
+          {renderItem(t('search'), 'always', t('search'), goToSearch, 'search', !peeking && categoriesAvailable)}
+          {renderItem(
+            t('changeLanguage'),
+            'always',
+            t('changeLanguage'),
+            goToLanguageChange,
+            'language',
+            !!(!peeking && goToLanguageChange)
+          )}
+          {showShare && renderItem(t('share'), 'never', t('share'), onShare, undefined)}
+          {showChangeLocation && renderItem(t('changeLocation'), 'never', t('changeLocation'), goToLanding, undefined)}
+          {renderItem(t('settings'), 'never', t('settings'), goToSettings, undefined)}
+          {routeCityModel && renderItem(t('disclaimer'), 'never', t('disclaimer'), goToDisclaimer, undefined)}
```

Comment:
Do we really need to pass `undefined` here?

Reply:
no :)